<a href="https://colab.research.google.com/github/mariajosevezga24-hash/PROGRAMACION/blob/CLASES/Quiz_26_de_marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [167]:
%%writefile pedirnotas.py
def pedir_nombre():
  return input("Ingrese su nombre: ")

def pedir_nota():
  n1 = float(input("Nota 1: "))
  n2 = float(input("Nota 2: "))
  n3 = float(input("Nota 3: "))
  n4 = float(input("Nota 4: "))
  return [n1, n2, n3, n4]

Overwriting pedirnotas.py


In [168]:
%%writefile promedio.py
import statistics
def calcular_promedio(n1, n2, n3, n4):
    return(n1 + n2 + n3 + n4) / 4

def mediana(n1, n2, n3, n4):
    return statistics.median([n1, n2, n3, n4])

def moda(n1, n2, n3, n4):
    return statistics.mode([n1, n2, n3, n4])

Overwriting promedio.py


In [169]:
import pedirnotas
nombre = pedirnotas.pedir_nombre()

Ingrese su nombre: Majo


In [165]:
import importlib
import pedirnotas
importlib.reload(pedirnotas)
notas = pedirnotas.pedir_nota()

Nota 1: 4.5
Nota 2: 4.6
Nota 3: 4.7
Nota 4: 4.7


In [170]:
notas = pedirnotas.pedir_nota()

Nota 1: 4.5
Nota 2: 4.6
Nota 3: 4.7
Nota 4: 4.8


In [180]:
import promedio
import importlib
importlib.reload(promedio)


prom = promedio.calcular_promedio(notas[0], notas[1], notas[2], notas[3])
mediana = promedio.mediana(notas[0], notas[1], notas[2], notas[3])
moda = promedio.moda(notas[0], notas[1], notas[2], notas[3])

print("Estudiante:", nombre)
print("Promedio:", prom)
print("Mediana:", mediana)
print("Moda:", moda)

if prom >= 3:
    print("Aprobó")
else:
    print("Reprobó")





Estudiante: Majo
Promedio: 4.65
Mediana: 4.65
Moda: 4.5
Aprobó


Mediana Aritmetica : En promedio el estudiante obtuvo 4.65
Mediana . El valor quedivide el 50% de las calificaciones es 4.65
La calificación mas repitida es 4.5
El desempeño es bueno porque aprobó y supera la moda  